### ISSUE TITLE
volumemodifier container does not start

### ISSUE THREAD

**Issue by ethan-steinman-j1** [kubernetes-sigs/aws-ebs-csi-driver#1618]
/kind bug

**What happened?**
I'm migrating from the EKS addon version of the driver at 1.17.0 to a Helm managed version of the driver at 1.19.0.  I enabled the new volumemodifier feature, and that container never properly starts.
These are the logs output by the container:
```bash
kubectl logs -n kube-system ebs-csi-controller-6c4dc95577-fg4tb -c volumemodifier
exec /bin/volume-modifier-for-k8s: exec format error
```

**What you expected to happen?**
The volumemodifier container starts when enabled.

**How to reproduce it (as minimally and precisely as possible)?**
Chart version: 2.19.0
values.yaml:
```yaml
controller:
  volumeModificationFeature:
    enabled: true
  serviceAccount:
    annotations:
      eks.amazonaws.com/role-arn: ${role_arn}
```

**Anything else we need to know?**:

**Environment**
- Kubernetes version (use `kubectl version`): 
```
Server Version: version.Info{Major:"1", Minor:"26+", GitVersion:"v1.26.4-eks-0a21954", GitCommit:"4a3479673cb6d9b63f1c69a67b57de30a4d9b781", GitTreeState:"clean", BuildDate:"2023-04-15T00:33:09Z", GoVersion:"go1.19.8", Compiler:"gc", Platform:"linux/amd64"}
```
- Driver version: 1.19.0

**Comment by rdpsin** [comment]
Could you provide more details? I'm unable to reproduce it.


```
$ eksctl utils associate-iam-oidc-provider --region=$REGION --cluster=$NAME --approve
$ eksctl create iamserviceaccount --cluster $NAME --region $REGION --namespace kube-system --name ebs-csi-controller-sa --attach-policy-arn arn:aws:iam::aws:policy/service-role/AmazonEBSCSIDriverPolicy --override-existing-serviceaccounts  --approve
$ helm repo add aws-ebs-csi-driver https://kubernetes-sigs.github.io/aws-ebs-csi-driver
$ helm upgrade --install aws-ebs-csi-driver --namespace=kube-system --set controller.serviceAccount.create=false --set node.serviceAccount.create=false --set node.serviceAccount.name=ebs-csi-controller-sa --set controller.serviceAccount.name=ebs-csi-controller-sa --set controller.volumeModificationFeature.enabled=true aws-ebs-csi-driver/aws-ebs-csi-driver
$ k logs deployments/ebs-csi-controller -n kube-system -c volumemodifier
Found 2 pods, using pod/ebs-csi-controller-6ccfd5dfb7-vv6hx
I0526 15:20:18.156299       1 main.go:65] Version : v0.1.0
I0526 15:20:18.157537       1 common.go:111] Probing CSI driver for readiness
I0526 15:20:18.161582       1 main.go:102] CSI driver name: "ebs.csi.aws.com"
I0526 15:20:18.161938       1 leaderelection.go:245] attempting to acquire leader lease kube-system/volume-modifier-for-k8s-ebs-csi-aws-com...
I0526 15:20:18.175061       1 leaderelection.go:255] successfully acquired lease kube-system/volume-modifier-for-k8s-ebs-csi-aws-com
I0526 15:20:18.175217       1 leader_election.go:178] became leader, starting
I0526 15:20:18.175250       1 controller.go:93] "Starting external modifier" name="ebs.csi.aws.com"
```

**Comment by ethan-steinman-j1** [comment]
I continued troubleshooting on my end and realized the pods were scheduling onto arm64 based nodes.  I set up tolerations and nodeselectors to get them onto amd64 based nodes, and it starts up fine now.  Sorry for the false alarm.

**Comment by rdpsin** [comment]
Turns out, there was an error in our build process, causing arm64 volumemodifier images to not work correctly. We have fixed the issue, and release v0.1.1 which should be able to run on arm64 nodes. Thanks for bringing this to our notice.
